### PR TITLE
feat(db): escapeLike/likePattern helpers + explicit dialect marker

### PR DESCRIPTION
**Pull request by forinda**
Two query-layer safety items from the audit, in kickjs-db.

## `escapeLike` / `likePattern`
User search text dropped into a `LIKE` pattern lets the user inject wildcards — searching `100%` becomes match-all, `a_b` matches `axb`, and a literal `%` under a leading wildcard can force a full scan. `escapeLike(input)` escapes `%`, `_`, and the escape char (backslash first, so no double-escaping); `likePattern(input, mode)` wraps it for `'contains'`/`'startsWith'`/`'endsWith'`/`'exact'`. Both exported from the root.

```ts
db.selectFrom('users').where('email', 'like', likePattern(ctx.qs().search, 'contains'))
```

## Explicit dialect marker
`detectDialect` previously relied solely on Kysely **ctor-name regex** (`/Postgres/i`) with a **silent fallback to SQLite** — a hand-rolled or future Kysely dialect whose ctor name didn't match was misclassified and emitted the wrong JSON-aggregation SQL. `pgDialect`/`mysqlDialect`/`sqliteDialect` now stamp a non-enumerable `KICK_DIALECT` marker (`Symbol.for`), and detection reads it first; the ctor-name heuristic stays as the fallback for raw Kysely dialects. `markDialect`/`readDialectMark` exported for adopters wrapping a raw dialect.

## Testing
- 11 new unit tests (escapeLike/likePattern edge cases incl. double-escape; marker stamp/read/non-enumerability + factory round-trip).
- Full db suite green: 77 files, 452 tests, typecheck enforced (`--typecheck`) clean.

## Deferred (not a polish item)
Cursor/keyset pagination — a real feature (stable ordering, cursor token encode/decode, API surface) that warrants its own scoped design, not a windup cram. Offset pagination is unchanged.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added LIKE/ILIKE query utilities for safe, literal pattern matching when building SQL queries with user input
  * Improved database dialect detection through explicit tagging, enhancing reliability across PostgreSQL, MySQL, and SQLite connections
  * New dialect utilities now available for integrations requiring explicit dialect identification

### CODE REVIEW

File: packages/db/src/query/like.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ */
+export function escapeLike(input: string, escapeChar = '\\'): string {
+  // Escape the escape char first so we don't double-escape the
+  // backslashes we add for % and _.
+  const e = escapeChar
+  return input
+    .replaceAll(e, e + e)
+    .replaceAll('%', e + '%')
+    .replaceAll('_', e + '_')
+}
```
[the remark this answers is not in the document]
Comment (forinda) [reply packages/db/src/query/like.ts:44]: Fixed in e24f1dfc — escapeLike now throws when escapeChar is empty, multi-character, or itself a wildcard ('%'/'_'). New test covers all four bad inputs.

File: packages/db/__tests__/unit/dialect-marker.test.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    expect(readDialectMark(d)).toBe('postgres')
+  })
+
+  it('the mark is non-enumerable (does not leak into spreads/JSON)', () => {
+    const d = markDialect({ createAdapter: () => ({}) }, 'sqlite')
+    expect(Object.keys(d)).toEqual(['createAdapter'])
+    expect(JSON.stringify(d)).toBe('{}')
+    // …but still readable via the symbol.
+    expect((d as Record<symbol, unknown>)[KICK_DIALECT]).toBe('sqlite')
+  })
```
[the remark this answers is not in the document]
Comment (forinda) [reply packages/db/__tests__/unit/dialect-marker.test.ts:23]: Fixed in e24f1dfc — the test now asserts a spread does NOT carry the symbol and checks Object.getOwnPropertyDescriptor(...).enumerable === false, since Object.keys/JSON.stringify ignore symbols regardless and didn't actually prove non-enumerability.

File: packages/db/__tests__/unit/dialect-marker.test.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  it('returns undefined for an unmarked dialect (ctor-name fallback territory)', () => {
+    expect(readDialectMark({})).toBeUndefined()
+  })
+
+  it('the factory-stamped tag survives — createDbClient detects it exactly', async () => {
+    const { sqliteDialect } = await import('../../src/adapters/sqlite/dialect')
+    // A fake better-sqlite3-shaped handle is enough; we only read the mark.
+    const dialect = sqliteDialect({ database: {} as never })
+    expect(readDialectMark(dialect)).toBe('sqlite')
+  })
```
[the remark this answers is not in the document]
Comment (forinda) [reply packages/db/__tests__/unit/dialect-marker.test.ts]: Renamed to 'the factory-stamped tag survives on dialect instances' (e24f1dfc) — the assertion only checks readDialectMark, so the name now matches. createDbClient's consumption of the marker is covered indirectly by the dialect-specific compile tests.

File: packages/db/src/query/like.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  switch (mode) {
+    case 'contains':
+      return `%${safe}%`
+    case 'startsWith':
+      return `${safe}%`
+    case 'endsWith':
+      return `%${safe}`
+    case 'exact':
+      return safe
+  }
```
[the remark this answers is not in the document]
Comment (forinda) [reply packages/db/src/query/like.ts:79]: Fixed in 2344f736 — added a default branch with a never-typed exhaustiveness check that throws on an unsupported mode (covers JS callers / as-any). Test added.

File: packages/db/src/dialect-marker.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * KickJS's own dialect factories (`pgDialect` / `mysqlDialect` /
+ * `sqliteDialect`) now stamp this marker, so detection is exact for the
+ * supported path. The ctor-name heuristic remains as the fallback for
+ * raw Kysely dialects an adopter constructs directly.
+ *
+ * `Symbol.for` so the marker is shared across module-identity boundaries
+ * (multiple copies of `@forinda/kickjs-db`).
+ */
+export const KICK_DIALECT: unique symbol = Symbol.for('@forinda/kickjs-db/dialect') as never
```
[the remark this answers is not in the document]
Comment (forinda) [reply packages/db/src/dialect-marker.ts]: Fixed in 2344f736 — but the suggested `as unknown as unique symbol` actually fails with TS2322: TypeScript only grants `unique symbol` to a const whose initializer is a *direct* Symbol()/Symbol.for() call, not a cast expression. Dropped the annotation and cast entirely — KICK_DIALECT is now plain `symbol` (sufficient for a computed property key), no unsafe cast at all. Verified with `tsgo -p tsconfig.json` exit 0.

File: packages/db/src/client/create.ts [hunk 6]
```diff
@@ -129,6 +130,13 @@ export function createDbClient<TSchema, DB = SchemaToTypes<TSchema>>(
 }
 
 function detectDialect(dialect: KyselyDialect): KickDbClient['dialect'] {
+  // Fast path: KickJS's own dialect factories (`pgDialect` /
+  // `mysqlDialect` / `sqliteDialect`) stamp an explicit marker, so
+  // detection is exact and never silently mis-classifies.
+  const marked = readDialectMark(dialect)
+  if (marked) return marked
```
[the remark this answers is not in the document]
Comment (forinda) [reply packages/db/src/client/create.ts:137]: Fixed in 2344f736 — readDialectMark now validates the marker value via isDialectTag() and returns undefined for anything that isn't a supported tag, so a rogue global-symbol stamp falls detectDialect back to ctor-name detection instead of an impossible dialect state. Test covers an 'oracle' stamp.
